### PR TITLE
Implemented lazy loading for resource tab

### LINKED ISSUE #770: Learning resource panel should not load all tabs content
Right now when the user opens the panel, there are 4 ajax requests to load the content of all the tabs.

Instead we should load only the content of the visible tab and load the others only if the tab become visible.

### PR DESCRIPTION
**Pull request by noisecapella**
Fixes #770

### CODE REVIEW

**Comment by jamiefolsom** [comment]
I'll look at this one.

**Comment by jamiefolsom** [comment]
- [x] Functionality
- [x] Code
- [x] Tests pass
- [x] Questions

**Comment by jamiefolsom** [comment]
@noisecapella -- should we set the active tab in the panel to the vocabulary tab every time we load the panel, so that XML is not loaded unless explicitly requested? 

Steps: First time I click on an LR, the panel opens, with vocab active. All good. Click on XML tab. Then close the panel. Then open another LR, and the XML tab is still selected. Might be we're loading that content too eagerly.

**Comment by noisecapella** [comment]
That's a good idea, I'll make that change

File: ui/static/ui/js/listing.js [hunk 1]
```diff
@@ -234,10 +234,49 @@ define('listing',
           closeLearningResourcePanel,
           $("#tab-1")[0]
         );
-        $('.cd-panel').addClass('is-visible');
-        StaticAssets.loader(repoSlug, resourceId, $("#tab-3")[0]);
+      };
```
Comment (jamiefolsom) [inline ui/static/ui/js/listing.js:237]: Nit: add a newline here.

File: ui/static/ui/js/listing.js [hunk 2]
```diff
@@ -234,10 +234,49 @@ define('listing',
...
+      var loadXmlTab = function(resourceId) {
         XmlPanel.loader(repoSlug, resourceId, $("#tab-2")[0]);
       };
```
Comment (jamiefolsom) [inline ui/static/ui/js/listing.js:241]: Nit: add a newline here.

**Comment by jamiefolsom** [comment]
General comment, perhaps unavoidable, but listing.js is getting hard to navigate, at almost 600 lines of js. Time to consider refactoring it?

**Comment by noisecapella** [comment]
Already doing it #797

**Comment by jamiefolsom** [comment]
Cool.

File: ui/static/ui/js/listing.js [hunk 3]
```diff
@@ -234,10 +234,49 @@ define('listing',
...
+      };
+
+      var loadedPanels;
+      var openResourcePanel = function(resourceId) {
+        // Reset loaded panels
+        loadedPanels = {};
+        currentResourceId = resourceId;
+
+        // Load the selected tab
+        _.each(panelLoaders, function(loader, tag) {
```
Comment (jamiefolsom) [inline ui/static/ui/js/listing.js]: Could this...

File: ui/static/ui/js/listing.js [hunk 4]
```diff
@@ -234,10 +234,49 @@ define('listing',
...
+          if ($("a[href='#" + tag + "']").attr("aria-expanded") === 'true') {
+            loader(resourceId);
+            loadedPanels[tag] = true;
+          }
+        });
+
+        $('.cd-panel').addClass('is-visible');
+      };
+
+      _.each(panelLoaders, function(loader, tag) {
```
Comment (jamiefolsom) [inline ui/static/ui/js/listing.js:270]: ...and this be refactored into a single function? Or are they too different?

**Comment by jamiefolsom** [comment]
All my notes are optional. Looks good and works well. :+1: from me.

**Comment by noisecapella** [comment]
Ok, it should now load the resource tab whenever the panel opens

**Comment by jamiefolsom** [comment]
Nice. Works well. :+1:

**Pull request merged**